File: src/utils/logger.py
"""Logger module providing a configured logger instance for the tool."""

import logging


class Logger:
    """Wraps Python's logging module with a pre-configured stream handler."""

    def __init__(self):
        """Set up the DDoSLogger with formatting."""
        self.logger = logging.getLogger('DDoSLogger')
        self.setup_logger()

    def setup_logger(self):
        """Attach a stream handler with timestamp/level formatting."""
        handler = logging.StreamHandler()
        formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
        handler.setFormatter(formatter)
        self.logger.addHandler(handler)

    def get_logger(self):
        """Return the configured logger instance."""
        return self.logger

File: src/utils/logger.rs
use log::{info, error};

pub fn setup_logger() {
    env_logger::init();
}

#[macro_export]
macro_rules! log {
    ($($t:tt)*) => {{
        if let Ok(level) = std::env::var("LOG_LEVEL") {
            if level == "error" {
                error!($($t)*);
            } else {
                info!($($t)*);
            }
        } else {
            info!($($t)*);
        }
    }}
}
